forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,223 @@
+    let ts = chain_head.0.min_timestamp();
+    let now = Instant::now().elapsed().as_secs();
+    let delta = ts - now;
+    let behind = delta;
+
+    let sync_status = if delta < EPOCH_DURATION_SECONDS as u64 * 3 / 2 {
+        // within 1.5 epochs
+        SyncStatus::Ok
+    } else if delta < 30 * 5 {
+        // within 5 epochs
+        SyncStatus::Slow
+    } else {
+        SyncStatus::Behind
+    };
+
+    let base_fee = chain_head.0.min_ticket_block().parent_base_fee().clone();
+
+    // chain health
+    let mut num_tipsets = 1;
+    let mut block_count = 0;
+    let blocks_per_tipset_last_finality = if epoch > chain_finality {
+        let mut ts = chain_head.0;
+
+        for _ in 0..chain_finality {
+            block_count += ts.blocks().len();
+            let tsk = ts.parents();
+            let tsk = TipsetKeysJson(tsk.clone());
+            if let Ok(tsjson) = chain_get_tipset((tsk,), &config.client.rpc_token).await {
+                ts = tsjson.0;
+                num_tipsets += 1;
+            }
+        }
+
+        block_count as f64 / chain_finality as f64
+    } else {
+        block_count as f64 / num_tipsets as f64
+    };
+
+    let health = 100. * blocks_per_tipset_last_finality / BLOCKS_PER_EPOCH as f64;
+
+    Ok(NodeStatusInfo {
+        behind,
+        health,
+        epoch,
+        base_fee,
+        sync_status,
+    })
+}
+
+impl InfoCommand {
+    pub async fn run(&self, config: Config, opts: &CliOpts) -> anyhow::Result<()> {
+        let NodeStatusInfo {
+            health,
+            behind,
+            epoch,
+            base_fee,
+            sync_status,
+        } = node_status(&config).await?;
+
+        // uptime
+        let start_time = start_time(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let network = chain_get_name((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        // Wallet info
+        let default_wallet_address = if let Some(default_wallet_address) =
+            wallet_default_address((), &config.client.rpc_token)
+                .await
+                .map_err(handle_rpc_err)?
+        {
+            Some(default_wallet_address)
+        } else {
+            None
+        };
+
+        display_info(
+            health,
+            start_time,
+            sync_status,
+            &network,
+            behind,
+            epoch,
+            base_fee,
+            default_wallet_address,
+            &opts.color,
+        )?;
+
+        Ok(())
+    }
+}
+
+fn display_info(
+    health: f64,
+    start_time: OffsetDateTime,
+    sync_status: SyncStatus,
+    network: &str,
+    behind: u64,
+    epoch: i64,
+    base_fee: TokenAmount,
+    default_wallet_address: Option<String>,
+    color: &LoggingColor,
+) -> anyhow::Result<()> {
+    let use_color = match color {
+        LoggingColor::Auto | LoggingColor::Always => true,
+        LoggingColor::Never => false,
+    };
```

Comment:
This is buggy. Use `color.coloring_enabled()`.

Reply:
Fixed.